tests/test_utils.py

```diff
@@ -317,3 +319,170 @@ def check_hash_is_valid(dfh):
+
+    # ---- Test case 4 check loading parameters using read excel and read csv method produces same output
+    class DummyModule(Module):
+        """Dummy module that will test reading from :py:func:pandas.excel and :py:func:pandas.read_csv"""
+        METADATA = {Metadata.DISEASE_MODULE}
+        PARAMETERS = {
+            'int_basic': Parameter(Types.INT, 'int basic'),
+            'real_basic': Parameter(Types.REAL, 'int basic'),
+            'categorical_basic': Parameter(Types.CATEGORICAL, 'int basic', categories=['category_1']),
+            'list_basic_int': Parameter(Types.LIST, 'int basic'),
+            'list_basic_real': Parameter(Types.LIST, 'int basic'),
+            'string_basic': Parameter(Types.STRING, 'int basic'),
+            'bool_true': Parameter(Types.BOOL, 'int basic'),
+            'bool_false': Parameter(Types.BOOL, 'int basic'),
+            'date': Parameter(Types.DATE, 'int basic'),
+        }
+
+        def read_parameters(self, data_folder):
+            # test read converted csv files produces same dataframe as excel file
+            excel_file_path = Path(data_folder / 'ResourceFile_test_convert_to_csv.xlsx')
+            # convert the above Excel file into csv equivalent. we will use the newly converted files to determine if
+            # loading parameters from Excel file will be equal to loading parameters from the converted csv files
+            convert_excel_files_to_csv(folder=Path(resourcefilepath / 'ResourceFile_test_convert_to_csv'),
+                                       files=[excel_file_path.name], test_mode=True)
+
+            # get excel sheet names
+            xls = pd.ExcelFile(excel_file_path)
+            df_excel = pd.read_excel(xls, sheet_name=xls.sheet_names)
+
+            # read newly converted csv files using read_csv_files method
+            df_csv = read_csv_files(Path(str(excel_file_path).split('.')[0]),
+                                    files=['dummy_parameter_values', 'parameter_values'])
+
+            # dictionary keys from both dataframe dictionaries should match
+            assert df_excel.keys() == df_csv.keys()
+
+            # load parameters from read_excel dataframe
+            self.load_parameters_from_dataframe(df_excel['parameter_values'])
+            excel_param_dict = self.parameters
+
+            # reset parameters
+            self.parameters = {}
+
+            # load parameters from read_csv dataframe
+            self.load_parameters_from_dataframe(df_csv['parameter_values'])
+            csv_param_dict = self.parameters
+
+            # parameters dictionary from `pd.read_excel` and `pd.read_csv` dataframes should be the same
+            assert excel_param_dict == csv_param_dict
+
+        def initialise_population(self, population):
+            pass
+
+        def initialise_simulation(self, sim):
+            pass
+
+    dummy_module = DummyModule()
+    dummy_module.read_parameters(Path(resourcefilepath / 'ResourceFile_test_convert_to_csv'))
```

Comment:
Hi @matt-graham . The only challenge with testing for matching dataframes  outside a disease module is that at this stage `read_csv_files` is outputting datframes with all data types as strings. At the other end `pandas.read_excel` is doing all the necessary data type formatting as such the two dataframes though having same values are appearing to differ in datatypes. That;s why I opted for a Dummy module to show that even though this is the case with input, it will be fine in the real modules because all parameters will be formatted accordingly using parameter datatypes defined in their respective disease modules. I don't know if you get what I mean?

Reply:
Hi @mnjowe. Ah good point around the data types, I hadn't realised the pandas `read_excel` and `read_csv` functions differed in how they handled data types. I can see then why it would make sense to check that we get equivalent parameter values read in whether we use an Excel or CSV resource file. As this is more about testing the functionality of the `load_parameters_from_dataframe` method rather than `read_csv_files`it might make more sense to have a test of this form in `tests/test_core.py` given the `Module.load_parameters_from_dataframe` method is defined in `src/tlo/core.py`. 

Just to note we can at least check that the loaded dataframes match in value when converting all columns to `str` dtype by adding something like 
```Python
    # values in dataframes when cast to strings should match
    for key in df_excel:
        assert df_excel[key].astype(str).equals(df_csv[key].astype(str))
```
to the end of the current `test_read_csv_files_method` test function.[](url)

Reply:
Thanks Matt and thanks for your suggestion on how the dataframes from these two methods can be tested. I'm okay replacing dummy module with this check. I feel that if we can confirm two dataframes from these two methods are equal, then we can be assured of equal output when these are passed to load_parameters function.